### DISCUSSION TITLE
Support for `customColors` prop for `ui.colors`

### DISCUSSION THREAD

**Discussion by Kolterdyx** [zauberzeug/nicegui#3707] [Ideas / Feature Requests]
In the Quasar docs references by the `ui.colors` docs, there is a link to an adjacent page for the [Quasar Color Palette config docs](https://quasar.dev/style/color-palette), where it shows that a `customColors` prop exists. I haven't used Quasar personally myself, but I'd assume it allows you to define custom color classes and use them like the main colors (primary, secondary, accent, etc.).

My idea would be to extend the `ui.colors` functionality to support more arguments the following way:

```python
from nicegui import ui

ui.colors(myyellow='#ff0')
ui.button('I am yellow!', color='myyellow')

ui.run()
```

This canbe done manually by adding the required CSS and adding the color name to the `QUASAR_COLORS` set in `nicegui.elements.mixins.color_elements` before using it, but I think this is diving too deep into implementation details and should be avoided, and is instead more suited for a new feature.

I think this would be a nice feature to support to allow extended theming, and after tinkering a bit with the source, it would be really easy to implement, so I submitted a PR #3708

**Comment by falkoschindler** [comment]
PR #3708 has just been released. 🚀
